### PR TITLE
Fixed Misaligned text when there's no data

### LINKED ISSUE #8805: Misaligned Text - "No patients to show" in List View of Shifting Page
**Bug Description**

On the Shifting page, when a search query returns no results, the alignment of the "No Results Found" text differs from the alignment of the same message on other pages within the application. 

---

**Steps To Reproduce**

1. Go to **Shifting** tab
2. Select **List View**
3. Search For a Patient Who doesn't exist
4. See the text alignment and size, and compare it with how it is being displayed in other pages

---

**Expected behavior**

The text **No patients to show.** should be displayed properly in centre like how it's being displayed in other pages, to maintain consistency across the pages.

---

**Screenshots**

In **Assets** Page:

Image

In **Sample Test** Page:

Image

In **Shifting** Page:

Image

---

**Desktop**

- OS: Windows
- Browser : Chrome
- Version: 129.0.6668.101

### PR DESCRIPTION
**Pull request by nithish1018**
## Proposed Changes

- Fixes #8805 
- Updated the rendering of the shifting card list
- The grid is only rendered when there's data to display.

This is how it looks now:

image
image

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

File: src/Components/Shifting/ListView.tsx [hunk 1]
```diff
@@ -60,171 +61,181 @@ export default function ListView() {
   });
 
   const showShiftingCardList = (data: any) => {
```
Comment (bodhish) [inline src/Components/Shifting/ListView.tsx]: Since we are refactoring it lets add strong types.

File: package-lock.json [hunk 2]
```diff
@@ -115,6 +115,7 @@
     "apps/care_livekit_fe": {
       "name": "care-livekit",
       "version": "0.0.1",
+      "extraneous": true,
```
Comment (bodhish) [inline package-lock.json]: why do we have changes in `package-lock.json` 🤔

**Review by bodhish — CHANGES_REQUESTED** [review]

File: src/Components/Shifting/ListView.tsx [hunk 3]
```diff
@@ -60,171 +61,181 @@ export default function ListView() {
...
-    if (data && !data.length) {
+    if (!data || data.length === 0) {
```
Comment (rithviknishad) [inline src/Components/Shifting/ListView.tsx:64]:
```suggestion
    if (!data?.length) {
```

Comment (rithviknishad) [inline src/Components/Shifting/ListView.tsx:71]: When `data` from useQuery is undefined, its likely that it's still loading. Show `<Loading />` in such situations. Can you confirm the same and show Loading if so?

File: src/Components/Shifting/ListView.tsx [hunk 4]
```diff
@@ -292,12 +303,13 @@ export default function ListView() {
...
-            <div>
-              <Pagination totalCount={shiftData?.count || 0} />
-            </div>
+            {showShiftingCardList(shiftData?.results || [])}
+
+            {(shiftData?.results?.length ?? 0) > 0 && (
+              <div>
+                <Pagination totalCount={shiftData?.count || 0} />
+              </div>
+            )}
```
Comment (rithviknishad) [inline src/Components/Shifting/ListView.tsx]: Is this check needed? Pagination component hides itself if the count is 0

image

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/Components/Shifting/ListView.tsx (hunk 4 above)
Comment (nithishsiliveru) [reply src/Components/Shifting/ListView.tsx]: Yes, that check is not required, it's working as expected without that check. I'll remove it

File: src/Components/Shifting/ListView.tsx (hunk 3 above)
Comment (nithishsiliveru) [reply src/Components/Shifting/ListView.tsx:71]: Ok, I'll add that

File: package-lock.json (hunk 2 above)
Comment (nithishsiliveru) [reply package-lock.json]: No idea, I didn't add it manually. I think that got updated after running `` npm run install-all`` 🤔

Comment (rithviknishad) [reply package-lock.json]: You can discard the package-lock.json changes if no packages has been added/removed/updated before making commits.

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

File: package-lock.json [hunk 5]
```diff
@@ -19930,4 +19930,4 @@
       }
     }
   }
-}
+}
```
Comment (bodhish) [inline package-lock.json]: why make this change?

**Review by bodhish — APPROVED** [review]

File: package-lock.json (hunk 5 above)
Comment (nithishsiliveru) [reply package-lock.json]: I took the package.lock.json file present in developer branch and replaced that with the one present in my branch

Comment (rithviknishad) [reply package-lock.json]: Line ending seems to be changed.

@bodhish should we add `.gitattributes` to always use LF line endings? https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings#per-repository-settings

Comment (sainak) [reply package-lock.json]: this change removed the new line at the end of the file which is not affected by line endings

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**